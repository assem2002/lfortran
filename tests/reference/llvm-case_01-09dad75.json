{
    "basename": "llvm-case_01-09dad75",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/case_01.f90",
    "infile_hash": "b0377013ea9328e6a8544ad056d49280c21a7b4c96cd9ee7670ddf0d",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-case_01-09dad75.stdout",
    "stdout_hash": "a972ffb6f87073b366ab41d8e24f62aa1e1a1b7d4090a40b077685c8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}