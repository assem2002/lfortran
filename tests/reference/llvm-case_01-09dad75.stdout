; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [2 x i8] c" \00", align 1
@1 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@2 = private unnamed_addr constant [2 x i8] c"1\00", align 1
@3 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@4 = private unnamed_addr constant [2 x i8] c" \00", align 1
@5 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@6 = private unnamed_addr constant [2 x i8] c"2\00", align 1
@7 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@8 = private unnamed_addr constant [2 x i8] c" \00", align 1
@9 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@10 = private unnamed_addr constant [2 x i8] c"3\00", align 1
@11 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@12 = private unnamed_addr constant [2 x i8] c" \00", align 1
@13 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@14 = private unnamed_addr constant [2 x i8] c"4\00", align 1
@15 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@16 = private unnamed_addr constant [11 x i8] c"ERROR STOP\00", align 1
@17 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@18 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@19 = private unnamed_addr constant [2 x i8] c" \00", align 1
@20 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@21 = private unnamed_addr constant [2 x i8] c"1\00", align 1
@22 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@23 = private unnamed_addr constant [2 x i8] c" \00", align 1
@24 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@25 = private unnamed_addr constant [6 x i8] c"2,3,4\00", align 1
@26 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@27 = private unnamed_addr constant [11 x i8] c"ERROR STOP\00", align 1
@28 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@29 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1

define i32 @main(i32 %0, i8** %1) {
.entry:
  %i = alloca i32, align 4
  %out = alloca i32, align 4
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  %i1 = alloca i32, align 4
  %out2 = alloca i32, align 4
  store i32 4, i32* %i1, align 4
  %2 = load i32, i32* %i1, align 4
  %3 = icmp eq i32 %2, 1
  br i1 %3, label %then, label %else

then:                                             ; preds = %.entry
  store i32 10, i32* %out2, align 4
  %4 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @2, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @3, i32 0, i32 0), i8* %4, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @1, i32 0, i32 0))
  br label %ifcont11

else:                                             ; preds = %.entry
  %5 = load i32, i32* %i1, align 4
  %6 = icmp eq i32 %5, 2
  br i1 %6, label %then3, label %else4

then3:                                            ; preds = %else
  store i32 20, i32* %out2, align 4
  %7 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @6, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @7, i32 0, i32 0), i8* %7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @5, i32 0, i32 0))
  br label %ifcont10

else4:                                            ; preds = %else
  %8 = load i32, i32* %i1, align 4
  %9 = icmp eq i32 %8, 3
  br i1 %9, label %then5, label %else6

then5:                                            ; preds = %else4
  store i32 30, i32* %out2, align 4
  %10 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @10, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @11, i32 0, i32 0), i8* %10, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @9, i32 0, i32 0))
  br label %ifcont9

else6:                                            ; preds = %else4
  %11 = load i32, i32* %i1, align 4
  %12 = icmp eq i32 %11, 4
  br i1 %12, label %then7, label %else8

then7:                                            ; preds = %else6
  store i32 40, i32* %out2, align 4
  %13 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @14, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @15, i32 0, i32 0), i8* %13, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @13, i32 0, i32 0))
  br label %ifcont

else8:                                            ; preds = %else6
  br label %ifcont

ifcont:                                           ; preds = %else8, %then7
  br label %ifcont9

ifcont9:                                          ; preds = %ifcont, %then5
  br label %ifcont10

ifcont10:                                         ; preds = %ifcont9, %then3
  br label %ifcont11

ifcont11:                                         ; preds = %ifcont10, %then
  %14 = load i32, i32* %out2, align 4
  %15 = icmp ne i32 %14, 40
  br i1 %15, label %then12, label %else13

then12:                                           ; preds = %ifcont11
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @18, i32 0, i32 0), i8* getelementptr inbounds ([11 x i8], [11 x i8]* @16, i32 0, i32 0), i8* getelementptr inbounds ([2 x i8], [2 x i8]* @17, i32 0, i32 0))
  call void @exit(i32 1)
  br label %ifcont14

else13:                                           ; preds = %ifcont11
  br label %ifcont14

ifcont14:                                         ; preds = %else13, %then12
  %16 = load i32, i32* %i1, align 4
  %17 = icmp eq i32 %16, 1
  br i1 %17, label %then15, label %else16

then15:                                           ; preds = %ifcont14
  store i32 11, i32* %out2, align 4
  %18 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @21, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @22, i32 0, i32 0), i8* %18, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @20, i32 0, i32 0))
  br label %ifcont20

else16:                                           ; preds = %ifcont14
  %19 = load i32, i32* %i1, align 4
  %20 = icmp eq i32 %19, 2
  %21 = load i32, i32* %i1, align 4
  %22 = icmp eq i32 %21, 3
  %23 = icmp eq i1 %20, false
  %24 = select i1 %23, i1 %22, i1 %20
  %25 = load i32, i32* %i1, align 4
  %26 = icmp eq i32 %25, 4
  %27 = icmp eq i1 %24, false
  %28 = select i1 %27, i1 %26, i1 %24
  br i1 %28, label %then17, label %else18

then17:                                           ; preds = %else16
  store i32 22, i32* %out2, align 4
  %29 = call i8* (i32, i8*, ...) @_lcompilers_string_format_fortran(i32 2, i8* null, i32 7, i8* getelementptr inbounds ([6 x i8], [6 x i8]* @25, i32 0, i32 0))
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @26, i32 0, i32 0), i8* %29, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @24, i32 0, i32 0))
  br label %ifcont19

else18:                                           ; preds = %else16
  br label %ifcont19

ifcont19:                                         ; preds = %else18, %then17
  br label %ifcont20

ifcont20:                                         ; preds = %ifcont19, %then15
  %30 = load i32, i32* %out2, align 4
  %31 = icmp ne i32 %30, 22
  br i1 %31, label %then21, label %else22

then21:                                           ; preds = %ifcont20
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @29, i32 0, i32 0), i8* getelementptr inbounds ([11 x i8], [11 x i8]* @27, i32 0, i32 0), i8* getelementptr inbounds ([2 x i8], [2 x i8]* @28, i32 0, i32 0))
  call void @exit(i32 1)
  br label %ifcont23

else22:                                           ; preds = %ifcont20
  br label %ifcont23

ifcont23:                                         ; preds = %else22, %then21
  br label %return

return:                                           ; preds = %ifcont23
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare i8* @_lcompilers_string_format_fortran(i32, i8*, ...)

declare void @_lfortran_printf(i8*, ...)

declare void @_lcompilers_print_error(i8*, ...)

declare void @exit(i32)
